### DISCUSSION TITLE
evaluate existing materials

### DISCUSSION THREAD

**Discussion by jhidding** [carpentries-incubator/efficient-computing-in-julia#6] [Ideas]
- [x] https://carpentries-incubator.github.io/julia-novice
- [x] https://github.com/carstenbauer/JuliaHLRS22 and https://github.com/carstenbauer/JuliaUCL24
- [x] https://juliafolds.github.io/data-parallelism/tutorials/quick-introduction/ 
- [x] https://juliateachingctu.github.io/Julia-for-Optimization-and-Learning/dev/ 
- [x] https://docs.julialang.org/en/v1/ 
- [x] https://www.matecdev.com/posts/julia-tutorial-science-engineering.html 
- [x] https://www.matecdev.com/posts/julia-numerical-computing.html 
- [x] https://www.matecdev.com/posts/julia-17x-faster-vs-python-scipy.html 
- [x] https://julialang.org/learning/tutorials/ (this is actually a list of materials, the relevant ones will be moved here)
- [x] [JuliaHub-Parallel Computing Guide (pdf)](https://info.juliacomputing.com/hubfs/JuliaHub-Parallel%20Computing%20Guide.pdf)
- [x] https://github.com/Arpeggeo/julia-distributed-computing
- [ ] Julia academy (this is mainly videos, are we considering video materials?)
- [x] https://techytok.com/from-zero-to-julia/
- [x] https://benlauwens.github.io/ThinkJulia.jl/latest/book.html
- [x] https://github.com/bkamins/The-Julia-Express
- [x] https://en.wikibooks.org/wiki/Introducing_Julia

**Comment by jhidding** [comment]
JuliaHub-Parallel computing guide is some nice documentation, but it doesn't use any guiding examples, so the material seems a bit too abstract for our purpose.

**Comment by jhidding** [comment]
https://juliafolds.github.io/data-parallelism/tutorials/quick-introduction/ 

Very nice example with Colatz conjecture.

**Comment by jhidding** [comment]
https://juliateachingctu.github.io/Julia-for-Optimization-and-Learning/dev/

High quality with nice examples

**Comment by abelsiqueira** [comment]
https://carpentries-incubator.github.io/julia-novice

Too random. Doesn't follow a good structure to teach. Gives advanced concepts before simple ones. Has complicated examples for basic things (e.g. loops).

**Comment by abelsiqueira** [comment]
https://github.com/carstenbauer/JuliaHLRS22

This is pretty much what we want. Day 2 is serial performance. Day 3 is parallelisation. Day 4 is GPU.

**Reply by abelsiqueira** [reply]
This seems to be a newer version: https://github.com/carstenbauer/JuliaUCL24

**Comment by abelsiqueira** [comment]
https://docs.julialang.org/en/v1/

Naturally, this is too extensive and broad, but it (or some specific pages) should be listed in the references.

**Comment by abelsiqueira** [comment]
https://www.matecdev.com/posts/julia-tutorial-science-engineering.html

This looks like a great introduction. It might be extensive for what we want, though, and too basic if we assume familiarity with programming.

**Comment by abelsiqueira** [comment]
https://www.matecdev.com/posts/julia-numerical-computing.html

These are essentially notes on Numerical computing using Julia. Not what we are looking for.

**Comment by abelsiqueira** [comment]
https://www.matecdev.com/posts/julia-17x-faster-vs-python-scipy.html

This might work as an example for why Julia, but it is mathematically complex. It shows an example of Curve Fitting with Splines in Python+Scipy and Julia+some packages. I would assume that it is too specific.

**Comment by abelsiqueira** [comment]
https://github.com/Arpeggeo/julia-distributed-computing

This could be relevant material if it was clearer and kept up to date. They set out to provide examples of how to do distributed computing correctly, but information is scarce.

**Comment by abelsiqueira** [comment]
https://benlauwens.github.io/ThinkJulia.jl/latest/book.html

Really introductory.

**Comment by abelsiqueira** [comment]
https://github.com/bkamins/The-Julia-Express

There as a PDF for 1.8. This looks very useful for us. It gives a lot of code examples instead of explanation, which might be suited for experienced programmers in other languages.

**Comment by abelsiqueira** [comment]
https://en.wikibooks.org/wiki/Introducing_Julia

Another good material, but too basic and extensive.
